File: common/src/main/java/com/yekong/common/base/BasePresenter.java
package com.yekong.common.base;

import android.content.Context;

/**
 * Created by xigua on 2017/10/16.
 */

public abstract class BasePresenter<T extends BaseAction,M extends BaseModel>{
    public M model;
    public T action;
    public Context context;
    public void setAction(T a,M m){
        this.action = a;
        this.model = m;
    }

}

File: kotlin_common/src/main/java/com/yekong/kotlin_common/base/BasePresenter.kt
package com.yekong.kotlin_common.base

import android.content.Context

/**
 * Created by xigua on 2017/10/19.
 */
class BasePresenter<A : BaseAction , M : BaseModel> {
    var a : A = null!!
    var m : M = null!!
    var context : Context = null!!

    fun setPresenter(a : A, m : M ,c : Context){
        this.a = a
        this.m = m
        this.context = c
    }
}